plugins/org.eclipse.n4js/src/org/eclipse/n4js/projectModel/IN4JSCore.java

```diff
@@ -165,4 +165,14 @@
 	 */
 	TModule loadModuleFromIndex(ResourceSet resourceSet, IResourceDescription resourceDescription,
 			boolean allowFullLoad);
+
+	/**
+	 * Converts a given URI to a platform or file URI so that:
+	 * <ul>
+	 * <li>All resources of user workspace projects are platform URIs.
+	 * <li>All resources of node_modules folders or shipped libraries are file URIs. The folder 'node_modules' itself is
+	 * only converted to a file URI if it is not inside a user workspace project.
+	 * </ul>
```

Comment:
Which of the two statements has priority, i.e. what happens if a resource is both in a user workspace project and inside a node_modules folder?
What about the headless case? (IN4JSCore is intended for both UI and headless)